target/linux/ath79/generic/base-files/lib/functions/mikrotik.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+#
+# Copyright (C) 2019 Robert Marko <robimarko@gmail.com>
+# Copyright (C) 2019 Roger Pueyo Centelles <roger.pueyo@guifi.net>
+#
+# Helper function to extract MAC addresses and calibration data for MikroTik
+#
+mikrotik_caldata_extract() {
+	local part=$1
+	local offset=$(($2))
+	local count=$(($3))
+	local firmware=$4
+	local mtd
+	local erdfile="/lib/firmware/erd.bin"
+
+	mtd=$(find_mtd_chardev $part)
+	[ -n "$mtd" ] || caldata_die "no mtd device found for partition $part"
+
+	rbextract -e $mtd $erdfile
+
+	dd if=$erdfile of=/lib/firmware/$firmware iflag=skip_bytes bs=$count skip=$offset count=1 2>/dev/null || \
```

Comment:
As stated in the subsequent patch, just hard-code /lib/firmware/$FIRMWARE here, as it's done with the other caldata extraction scripts.

Reply:
I think I can't do that, if I understood your comment correctly. This helper function is used by both ath9k and ath10k, which will use different firmware locations (e.g., /lib/firmware/ath9k-eeprom-ahb-18100000.wmac.bin and /lib/firmware/ath10k/cal-pci-0000:00:00.0.bin).

This is why, in ath9k/ath10k-eeprom files I was using

Reply:
As I understand it, the $FIRMWARE variable is evaluated when the function is called. Thus, when called for ath9k, $FIRMWARE will be "ath9k-eeprom-ahb-18100000.wmac.bin" and for ath10k, $FIRMWARE will be "ath10k/cal-pci-0000:00:00.0.bin". If not, we could use caldata_extract for both, too:
https://github.com/openwrt/openwrt/blob/master/package/base-files/files/lib/functions/caldata.sh#L20

Reply:
OK, this is also fixed.